soccer_ipm/soccer_ipm/soccer_ipm.py

```diff
@@ -0,0 +1,323 @@
+
+            # Check if post is not going out of the image at the bottom
+            if not self._object_at_bottom_of_image(
+                    self._bb_footpoint(robot.bb).y,
+                    self._goalpost_footpoint_out_of_image_threshold):
+                # Create footpoint
+                footpoint = Point2DStamped(
+                    header=msg.header,
+                    point=self._bb_footpoint(robot.bb)
+                )
+                # Map point from image onto field plane
+                try:
+                    relative_foot_point = self.ipm.map_point(
+                        field,
+                        footpoint,
+                        output_frame=self._base_footprint_frame)
+
+                    transformed_robot = sv3dm.Robot()
+                    transformed_robot.attributes = robot.attributes
+                    transformed_robot.confidence = robot.confidence
+                    transformed_robot.bb.center.position = relative_foot_point.point
+                    transformed_robot.bb.size.x = 0.3   # TODO better size estimation
+                    transformed_robot.bb.size.y = 0.3   # TODO better size estimation
+                    transformed_robot.bb.size.z = 0.5   # TODO better size estimation
+                    robots.robots.append(transformed_robot)
+                except NoIntersectionError:
+                    self.get_logger().warn(
+                        'Got a robot with foot point ({},{}) I could not transform.'.format(
+                            footpoint.point.x,
+                            footpoint.point.y),
+                        throttle_duration_sec=5)
+
+        self.robots__pub.publish(robots)
+
+    def callback_field_boundary(self, msg: FieldBoundary):
+        field = self.get_field(msg.header.stamp, 0.0)
+
+        field_boundary = sv3dm.FieldBoundary()
+        field_boundary.header.stamp = msg.header.stamp
+        field_boundary.header.frame_id = self._base_footprint_frame
+        field_boundary.confidence = field_boundary.confidence
+
+        for p in msg.points:
+            image_point = Point2DStamped(
+                header=msg.header,
+                point=p)
+            # Map point from image onto field plane
+            try:
+                relative_foot_point = self.ipm.map_point(
+                    field,
+                    image_point,
+                    output_frame=self._base_footprint_frame)
+
+                field_boundary.points.append(relative_foot_point.point)
+            except NoIntersectionError:
+                self.get_logger().warn(
+                    'Got a field boundary point ({},{}) I could not transform.'.format(
+                        image_point.point.x,
+                        image_point.point.y),
+                    throttle_duration_sec=5)
+
+        self.field_boundary_pub.publish(field_boundary)
+
+    def callback_masks(self, msg: Image, publisher, encoding='8UC1', scale: float = 1.0):
```

Comment:
I agree that converting masks is independent of the soccer vision messages. 
But I don't think that this is a Bit-Bots specific use case. I think it is a more general use case and therefore might not fit together with the soccer vision msgs specific things. Projecting masks to a point cloud on a plane has many use cases, e.g. for ground markings in a warehouse scenario or lines on a street etc.. 
If you don't want it in this PR, I suggest moving it thing to it's own node in the ipm repo. Similarly to the ipm action etc. this is a more general way of interacting with the ipm library as projecting a generic mask is not associated with any specific use case and close to the library API itself (set of image space points <-> mask is mapped to point cloud in Cartesian space). 
Adding another node with its own tf buffer etc. is not so cool when we consider the performance issues as discussed in our blog post, but they would make the setup definitely more general.

Reply:
That sounds good, I think you're right that it could me made more general. In fact, the method we're talking about here wouldn't just work for masks, it'll work for any image and thatll be very useful to have in ros-sports/ipm.

Reply:
Moved to #3